pkg/operator/controller/nodeobservability/controller.go

```diff
@@ -313,7 +322,34 @@ func (r *NodeObservabilityReconciler) ensureNodeObservabilityDeleted(ctx context
 
 // machineConfigChangeRequested returns true, when a given NodeObservabilityType needs
 // machine config change. Only CrioKubeletNodeObservabilityType requires a MC change, false otherwise
-func machineConfigChangeRequested(nodeObs *operatorv1alpha2.NodeObservability) bool {
+func (r *NodeObservabilityReconciler) machineConfigChangeRequested(ctx context.Context, nodeObs *operatorv1alpha2.NodeObservability) bool {
+	// avoid the creation of a new NOMC if the CRIO profiling is already enabled in the rendered MC
+	mcpName := types.NamespacedName{Name: workerMCP}
+	mcp := &mcv1.MachineConfigPool{}
+	err := r.ClusterWideClient.Get(ctx, mcpName, mcp)
```

Comment:
`ClusterWideClient` is mainly only included for accessing resources across namespaces. Cluster scoped resources we should be fine accessing using the just `Client`.

Reply:
Addressed